tuning/policydrivencontroller/controllermetrics/metrics.py

```diff
@@ -0,0 +1,244 @@
+import numpy as np
+from transformers import TrainerState, IntervalStrategy
+from transformers.utils import logging
+from collections import deque
+import math
+import copy
+
+logger = logging.get_logger(__name__)
+
+class MetricHandler:
+    """Base class for the controller-metrics"""
+    def validate(self, training_args) -> bool:
+        pass
+    
+    def compute(self, training_state, training_args=None, metrics=None) -> dict:
+        pass   
+
+class WindowStepLoss(MetricHandler):
```

Comment:
Why not have the same function for both step and epoch, and take interval strategy as a parameter? We can call the class as something like `LossInWindow`